Diary/Controller/DiaryViewController.swift

```diff
@@ -125,8 +167,105 @@ class DiaryViewController: UIViewController {
     }
     
     @objc private func saveDiary() {
-        setDiary()
-        update(diary)
-        delegate?.updateView()
+        if diary == nil {
+            if locationManager?.authorizationStatus == .authorizedAlways || locationManager?.authorizationStatus == .authorizedWhenInUse {
+                setWeatherInfo()
+            } else {
+                setDiary()
+            }
+        } else {
+            editDiary()
+        }
+    }
+}
+
+// MARK: - weather info
+extension DiaryViewController: CLLocationManagerDelegate {
+    private func getCoordinate() -> (latitude: Double, longitude: Double)? {
+        let coordinate = locationManager?.location?.coordinate
+        guard let latitude = coordinate?.latitude else {
+            return nil
+        }
+        
+        guard let longtitude = coordinate?.longitude else {
+            return nil
+        }
```

Comment:
guard문은 하나로 합쳐도 될 것 같아요.
```suggestion
        guard let latitude = coordinate?.latitude,
              let longtitude = coordinate?.longitude 
        else { return nil }
```

Reply:
guard 문 하나로 합칠 수 있는 부분 합쳤습니다~!

```swift
guard let latitude = coordinate?.latitude,
    let longitude = coordinate?.longitude else {
    return (0, 0)
}

guard let main = weatherData["main"] as? String,
    let iconID = weatherData["icon"] as? String else {
    return
}
```